lib/puppet/http/service.rb

```diff
@@ -35,8 +35,9 @@ def with_base_url(path)
     u
   end
 
-  def connect(ssl_context: nil)
-    @client.connect(@url, ssl_context: ssl_context)
+  def connect(options: {})
+    ssl_context = options.fetch(:ssl_context, nil)
+    @client.connect(@url, options: {ssl_context: ssl_context})
```

Comment:
Maybe just pass `options` directly? I could imagine being surprised if `Service#connect` is called with additional options in the future, but only `ssl_context` makes it through.

Reply:
What about the methods in the different resolvers? Should those accept a generic options hash too and pass it through?

Reply:
Oh sorry, I'd keep `Service#connect` as it was, accepting an `ssl_context: nil`, and have it pass `options: { ssl_context: ssl_context }` like you're doing.

I think only the low-level API `Client#{connect, get, etc}`) needs to accept the generic `options`.

Reply:
updated